src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php

```diff
@@ -762,4 +764,54 @@ public function testAdderAndRemoverArePreferredOverSetterForSameSingularAndPlura
 
         $this->assertEquals(['aeroplane'], $object->getAircraft());
     }
+
+    /**
+     * @expectedException \Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException
+     * @expectedExceptionMessageRegExp /.*The add method "addFoo" in class "Symfony\\Component\\PropertyAccess\\Tests\\Fixtures\\TestAdderRemoverInvalidMethods" was found, but the corresponding remove method "removeFoo" was not found\./
+     */
+    public function testAdderWithoutRemover()
+    {
+        $object = new TestAdderRemoverInvalidMethods();
+        $this->propertyAccessor->setValue($object, 'foos', [1, 2]);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException
+     * @expectedExceptionMessageRegExp /.*The remove method "removeBar" in class "Symfony\\Component\\PropertyAccess\\Tests\\Fixtures\\TestAdderRemoverInvalidMethods" was found, but the corresponding add method "addBar" was not found\./
+     */
+    public function testRemoverWithoutAdder()
+    {
+        $object = new TestAdderRemoverInvalidMethods();
+        $this->propertyAccessor->setValue($object, 'bars', [1, 2]);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException
+     * @expectedExceptionMessageRegExp /.*The method "addFoo" in class "Symfony\\Component\\PropertyAccess\\Tests\\Fixtures\\TestAdderRemoverInvalidArgumentLength" requires at least "1" parameters, "0" found. The method "removeFoo" in class "Symfony\\Component\\PropertyAccess\\Tests\\Fixtures\\TestAdderRemoverInvalidArgumentLength" requires at least "1" parameters, "2" found\./
+     */
+    public function testAdderAndRemoveNeedsTheExactParametersDefined()
+    {
+        $object = new TestAdderRemoverInvalidArgumentLength();
+        $this->propertyAccessor->setValue($object, 'foo', [1, 2]);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException
+     * @expectedExceptionMessageRegExp /.*The method "setBar" in class "Symfony\\Component\\PropertyAccess\\Tests\\Fixtures\\TestAdderRemoverInvalidArgumentLength" requires at least "1" parameters, "2" found./
```

Comment:
Such message doesn't make sense to me. Shouldn't it be reverse? Requires 2 arguments, 1 passed? Or is this when you pass more arguments than required? It's confusing, can we do something about it?

Reply:
The method needs to define only 1 parameter, but 2 are defined.
E.G

```
// This will trigger the error
public function setBar($var1, $var2) {}

// Instead it needs to be defined like
public function setBar($var1) {}
```

I'm not sure what the best wording is to describe this scenario, maybe `The method "setBar" in class "..." needs to define only "1" parameter. Currently, the method defines "2" parameters.`?

Reply:
"Method has to require 1 argument only, but requires 2"?

Reply:
Or "requires exactly 1 argument, but 2 are defined"?

Reply:
That's wrong though. Method doesn't require exactly 1 argument (currently), but it must require exactly 1 argument to function correctly.

Reply:
Indeed, that's the `exactly` that was worth interest imo, I shouldn't have put the rest :)

Reply:
Changed the message to `The method "setBar" requires 2 arguments, but should accept only 1`